framework/src/base/MooseApp.C

```diff
@@ -959,10 +961,16 @@ MooseApp::setupOptions()
 
     // Pass list of input files and optional text string if provided to parser
     if (!isParamValid("_input_text"))
-      _parser.parse(_input_filenames);
+    {
+      if (_input_filenames.empty())
+        mooseError("No input files specified. Add -i <inputfile> to your command line.");
```

Comment:
Did this error not exist before?

Reply:
This error used to be in Parser.C
```
if (_input_filenames.empty())
    mooseError("No input files specified.");
```
I moved it here because we want to allow empty input file or wrong input file type(e.g. input.e) in front parser and deal with all these options later in MooseApp.C, otherwise we will need add a lot of parameters like 
`"--show-outputs"`, `"--mesh-only [mesh_file_name]"`,` "--definition" `, etc. to new Parser along with the condition logic